src/sentry/tsdb/redis.py

```diff
@@ -381,6 +433,72 @@ def merge_aggregates(values):
             ]
         )
 
+    def merge_distinct_counts(self, model, destination, sources, timestamp=None):
+        rollups = {}
+        for rollup, samples in self.rollups.items():
+            _, series = self.get_optimal_rollup_series(
+                to_datetime(self.get_earliest_timestamp(rollup, timestamp=timestamp)),
+                end=None,
+                rollup=rollup,
+            )
+            rollups[rollup] = map(to_datetime, series)
+
+        temporary_id = uuid.uuid1().hex
+
+        def make_temporary_key(key):
+            return '{}{}:{}'.format(self.prefix, temporary_id, key)
+
+        data = {}
+        for rollup, series in rollups.items():
+            data[rollup] = {timestamp: [] for timestamp in series}
+
+        with self.cluster.fanout() as client:
+            for source in sources:
+                c = client.target_key(source)
+                for rollup, series in data.items():
+                    for timestamp, results in series.items():
+                        key = self.make_key(
+                            model,
+                            rollup,
+                            to_timestamp(timestamp),
+                            source,
+                        )
+                        results.append(c.get(key))
+                        c.delete(key)
```

Comment:
I know this is a convenient place to delete the keys, but maybe save this for after we've recorded the correct count? Not sure if it's worth it but there's a somewhat trivial amount of stuff in between deleting the source and updating the destination.